Don’t spawn multiple monitoring tasks for the reused coinbase tx

philipr-za opened the pull request:
## Description
Currently if multiple coinbase transactions are requested for the same block height the wallet will reuse the prior transaction but a whole new monitoring protocol task will be spawned for that coinbase tx even though one is already running. 

This PR adds TxId to be stored with the protocols monitoring base node responses and if one exists for a newly requested coinbase that is reusing a transaction a new monitoring protocol is not started.

## How Has This Been Tested?
Existing test cover this

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

sdbondi commented on base_layer/wallet/src/transaction_service/service.rs:
```diff
@@ -1729,6 +1730,17 @@ where
         }
 
+        // Confirm that an existing coinbase monitoring protocol doesn't already exist for this transaction
+        if self
+            .base_node_response_senders
+            .values()
+            .find(|(t, _s)| t == &tx_id)
+            .is_some()
```
```suggestion
        if self
            .base_node_response_senders
            .values()
            .any(|(t, _s)| t == &tx_id)
            
```

sdbondi commented on base_layer/wallet/src/transaction_service/service.rs:
```diff
@@ -1509,7 +1510,7 @@ where
                 return Ok(());
             },
-            Some(s) => s,
+            Some(s) => &mut s.1,
```
nit: 
```suggestion
            Some((_, s)) => s,
```

sdbondi reviewed (commented):
LGTM - some nits

philipr-za committed:
Don’t spawn multiple monitoring tasks for the reused coinbase tx

Currently if multiple coinbase transactions are requested for the same block height the wallet will reuse the prior transaction but a whole new monitoring protocol task will be spawned for that coinbase tx even though one is already running. 

This PR adds TxId to be stored with the protocols monitoring base node responses and if one exists for a newly requested coinbase that is reusing a transaction a new monitoring protocol is not started.

sdbondi reviewed (approved):
👍
